app/controllers/memberships_controller.rb

```diff
@@ -1,13 +1,35 @@
 class MembershipsController < ApplicationController
+  before_action :set_membership, only: %i[destroy update]
```

Comment:
Why not create `membership` as a memorised method then its only created and stored the first time its used instead of creating before_actions

```suggestion
  def membership
    @membership ||= Membership.find(params[:id])
  end
```